lib/conjur/command.rb

```diff
@@ -267,6 +267,92 @@ def display(obj, options = {})
         puts str
       end
 
+      def prompt_to_confirm kind, properties
+        puts
+        puts "A new #{kind} will be created with the following properties:"
+        puts
+        properties.select{|k,v| !v.blank?}.each do |k,v|
+          printf "%-10s: %s\n", k, v
+        end
+        puts
+        
+        exit(0) unless %w(yes y).member?(highline.ask("Proceed? (yes/no): ").strip.downcase)
+      end
+      
+      def integer? v
+        Integer(v, 10) rescue false
+      end
+    
+      def prompt_for_id kind, label = 'id'
+        highline.ask("Enter the #{label}: ") do |q|
+          q.readline = true
+          q.validate = lambda{|id|
+            !id.blank? && !api.send(kind, id).exists?
+          }
+          q.responses[:not_valid] = "A #{kind} called '<%= @answer %>' already exists"
+        end
+      end
+
+      def prompt_for_public_key
+        public_key = highline.ask("Enter the public key (press enter to skip): ") do |q|
+          q.validate = lambda{|key|
+            if key.blank?
+              true
+            else
+              validate_public_key key
+            end
+          }
+          q.responses[:not_valid] = "Public key format is invalid; please try again"
+        end
+        public_key.blank? ? nil : public_key.strip
+      end
+    
+      # http://serverfault.com/questions/453296/how-do-i-validate-a-rsa-ssh-public-key-file-id-rsa-pub
+      def validate_public_key key
+        if system('which ssh-keygen 2>&1 > /dev/null')
+          Conjur.log.debug "Using ssh-keygen to verify the public key\n" if Conjur.log
+          require 'tempfile'
+          tempfile = Tempfile.new 'public_key'
+          tempfile.write(key)
+          tempfile.close
+          `ssh-keygen -l -f #{tempfile.path}`
+          $? == 0
+        else
+          Conjur.log.debug "ssh-keygen is not available; falling back to simple string testing\n" if Conjur.log
+          # Should be a single line with 3 components
+          key, remainder = key.split("\n")
+          remainder.nil? && key.split(/\s+/).length == 3
+        end
+      end
+      
+      def prompt_for_group
+        group_ids = api.groups.map(&:id)
+        
+        highline.ask('Enter the group which will own the record (press enter to own the record yourself): ', [ "" ] + group_ids) do |q|
```

Comment:
There's a problem with that parenthetical: it's not actually true when creating a user (it will be owned by itself).